code/script.js

```diff
@@ -254,72 +271,165 @@ const checkQuestion = () => {
+  if (numberOfGuesses < 1) {
+    //counter for numberOfGuesses
+    window.alert('No more choices, please try again or make a guess!');
+    return;
+  }
+
+  numberOfGuesses--;
+  counter.innerText = numberOfGuesses;
+
   const { category, value } = currentQuestion
   // Show the correct alert message for different categories
+  // filter by category to keep or remove based on the keep variable.
+  // Invoke a function to redraw the board with the remaining people.
   if (category === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person wears ${value}! Keep all people that wears ${value}. !YOU HAVE 3 CHOICES!`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person['accessories'].includes(value))
+
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${value}! Remove all people that wears ${value}.  !YOU HAVE 3 CHOICES!`
       )
+      charactersInPlay = charactersInPlay.filter((person) => !person['accessories'].includes(value))
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person is a ${value}! Keep all people that are ${value}. !YOU HAVE 3 CHOICES!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person['other'].includes(value))
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person is not a ${value}! Remove all people that are ${value}.  !YOU HAVE 3 CHOICES!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person['other'].includes(value))
+    }
+  } else if (category === 'hair') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} hair! Keep all people with ${value} hair. !YOU HAVE 3 CHOICES!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person['hair'] === value)
+    } else {
+      alert(
+        `No, the person doesnt have ${value} hair! Remove all people with ${value} hair.  !YOU HAVE 3 CHOICES!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person['hair'] !== value)
+    }
+  } else if (category === 'eyes') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} eye! Keep all people with ${value} eye. !YOU HAVE 3 CHOICES!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person['eyes'] === value)
+    } else {
+      alert(
+        `No, the person doesnt have ${value} eye! Remove all people with ${value} eye.  !YOU HAVE 3 CHOICES!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person['eyes'] !== value)
```

Comment:
An alternative to having eyes and hair separate is to also use the 'category' in the sentence. Below is an example :)

if (keep) {
      alert(
        `Yes, the person has ${value} ${category}! Keep all the people that has ${value} ${category}`
      ); 
      charactersInPlay = charactersInPlay.filter((person) => 
        person[category] === value
      );
    } else {
      alert (
        `No, the person doesn't have ${value} ${category}! Remove all people that has ${value} ${category}`
      );
      charactersInPlay = charactersInPlay.filter((person) => 
        person[category] !== value);
};
};